src/pheromones/common/include/common/IProtobufStream.h

```diff
@@ -0,0 +1,39 @@
+#ifndef __IPROTOBUFSTREAM_H_
+#define __IPROTOBUFSTREAM_H_
+
+#include <cstdint>
+
+/**
+ *@brief An interface to manage the stream of data used to encode/decode.
+ *Used in the other pheromones libraries. This is the user job to implement this interface. The
+ *other classes uses this interface to get the data to deserialize and send the data to deserialize
+ **/
+class IProtobufStream {
+
+  public:
+    virtual ~IProtobufStream() = default;
+
+    /**
+     *@brief Used to receive the data to serialize
+     *
+     *@param [out] data buffer where the received data will be stored
+     *
+     *@param [in] length maxmimum length of the data to store in the buffer
+     *
+     *@return true if the operation was successful, false if not
+     **/
+    virtual bool receive(uint8_t* data, uint16_t length) = 0;
```

Comment:
Considering sockets have a semi-blocking recv which can return even without the full amout of bytes, should we do something similar here? This would probably also simplify the drivers that use a circular buffer. If we start near the end of the buffer, we could return up to the modulo point and then the next call would start from the beginning of the buffer

Reply:
Never mind what I said. The stream is not implemented, so this behaviour can be added in the implementation.